pkg/fixtures/fixtures.go

```diff
@@ -497,33 +275,81 @@ func (fxt *Fixture) updateEnv(env map[string]string) error {
 	return nil
 }
 
-// toFixtureQuery will parse a string into a fixtureQuery struct, additionally
-// returning a bool indicating the value did contain a fixtureQuery.
-func toFixtureQuery(value string) (fixtureQuery, bool) {
-	var query fixtureQuery
-	isQuery := false
+// isNameIn will search if the current fixture is in the skip list
+func isNameIn(name string, skip []string) bool {
+	for _, skipName := range skip {
+		if name == skipName {
+			return true
+		}
+	}
 
-	if r, didMatch := matchFixtureQuery(value); didMatch {
-		isQuery = true
-		match := r.FindStringSubmatch(value)
-		query = fixtureQuery{Name: match[1], Query: match[2], DefaultValue: match[3]}
+	return false
+}
+
+// buildRewrites takes a slice of json queries and values then builds
+// them into a map to later by merged. We work through the entire
+// list at the same time because the user might pass in multiple
+// changes for the same fixture.
+//
+// The query supported is <fixture_name>:path.to.field=value
+func buildRewrites(changes []string, toRemove bool) map[string]interface{} {
+	builtChanges := make(map[string]interface{})
+	for _, change := range changes {
+		changeSplit := strings.SplitN(change, "=", 2)
+		path := changeSplit[0]
+
+		// When removing a field there will be no value so we set a default
+		// empty string or trying to get the split value from above
+		var value string
+		if !toRemove {
+			value = changeSplit[1]
+		}
+
+		pathSplit := strings.SplitN(path, ":", 2)
+		name := pathSplit[0]
+		keys := pathSplit[1]
+
+		keysSplit := strings.Split(keys, ".")
+
+		keysReversed := reverse(keysSplit)
+		key, keysReversed := pop(keysReversed)
+		keyMap := make(map[string]interface{})
+		keyMap[key] = value
+
+		for _, key := range keysReversed {
+			keyMap = map[string]interface{}{
+				key: keyMap,
+			}
+		}
+		_, ok := builtChanges[name]
+		if ok {
+			if err := mergo.Merge(builtChanges[name], keyMap); err != nil {
+				fmt.Println(err)
+			}
+		} else {
+			builtChanges[name] = keyMap
+		}
 	}
 
-	return query, isQuery
+	return builtChanges
 }
 
-// matchQuery will attempt to find matches for a fixture query pattern
-// returning a *Regexp which can be used to further parse and a boolean
-// indicating a match was found.
-func matchFixtureQuery(value string) (*regexp.Regexp, bool) {
-	// Queries will start with `${` and end with `}`. The `:` is a
-	// separator for `name:json_path`. Additionally, default value will
-	// be specified after the `|`.
-	// example: ${name:json_path|default_value}
-	r := regexp.MustCompile(`\${([^\|}]+):([^\|}]+)\|?([^/\n]+)?}`)
-	if r.Match([]byte(value)) {
-		return r, true
+// pop returns the first item and the rest of the list minus the first item
+// From: https://github.com/golang/go/wiki/SliceTricks#pop
+func pop(list []string) (string, []string) {
+	return list[len(list)-1], list[:len(list)-1]
```

Comment:
@tomer-stripe i think this pops the last item, and return the list minus the last item? should we update the comment or code?

Reply:
@tomer-stripe also if we do want to pop the last item, do we still need to `reverse` it at line314?